torch/_dynamo/variables/tensor.py

```diff
@@ -40,6 +40,10 @@
 }
 
 
+def has_same_metadata(t1, t2):
+    return t1.size == t2.size and t1.stride == t2.stride
```

Comment:
not checking dtype/device?

Reply:
I'm not sure of those are mutable but I would check Storage, and storage offset as well

Reply:
Added dtype/device and storage offset checks.